### PR TITLE
Add list view and a markdown readme

### PR DESCRIPTION
**Pull request by bueltge**
Hi Shaun,

I use your url-shortener since a lot of years. Currently I need a list view and I enhance your script.
Also I add a readme in markdown format to read it much better on github.
Maybe you enjoy the enhancement.
Best.

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -4,6 +4,16 @@
...
 
+define('LESSN_VERSION',	'1.1.1');
+
+define( 'LESSN_ROOT', __DIR__ );
+define('LESSN_DOMAIN', 	preg_replace('#^www\.#', '', $_SERVER['SERVER_NAME']));
+define('LESSN_URL', 	str_replace('-/index.php', '', 'http://'.LESSN_DOMAIN.$_SERVER['PHP_SELF']));
+
+define('COOKIE_NAME', 	DB_PREFIX.'auth');
+define('COOKIE_VALUE',	md5(USERNAME.PASSWORD.COOKIE_SALT));
+define('COOKIE_DOMAIN', '.'.LESSN_DOMAIN);
```
Comment (jeffbyrnes) [inline index.php]: The addition of all of these constants is curious; how does this help with the list view? Is it truly necessary, or something that ended up in here that is related, but separate?

**Comment by jeffbyrnes** [comment]
:+1: on the addition of a list view, though some stuff in there seems unnecessary to me (probably just don’t fully understand the context or usage).

This can probably delete `README.txt` to avoid any confusion.

**Comment by bueltge** [comment]
@jeffbyrnes Thanks for the helping comments. I have centralized the constants. My idea was currently, not to much changes in the original source. Also now I have add a update readme and remove the readme.txt, replaced by the md-file. Maybe now is it more clear.

**Comment by jeffbyrnes** [comment]
@bueltge you’re welcome. Thanks for the explanation w/r/t the constants; perhaps an interactive rebase & some additional detail in the commit message would be helpful for the long term?

**Comment by bueltge** [comment]
Thanks. Which long term do you mean? Is a commit message from my side not clear enough, to much changes or wrong, not understandable commit message?

**Comment by jeffbyrnes** [comment]
@bueltge the commit message itself is what I meant; that could say more about what your intent is, and why, you did it that way.

Definitely like how you now have moved all the constants to their own, single file. Much better to have them just once!

Are you familiar with interactive rebasing?

**Comment by bueltge** [comment]
No, currently I'm not with the interactive rebasing. Now I use seldom this `git commit --amend`, not more.  Maybe you have a hint, link for me?
I think, I understand you goal, that the commit is clear for all readers there have interest, not to short and without details about why.

**Comment by jeffbyrnes** [comment]
@bueltge exactly that, in fact. Two resources to learn about interactive rebasing:
- [Thoughtbot’s “Git Interactive Rebase, Squash, Amend and Other Ways of Rewriting History”](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history)
- [Atlassian’s “Rewriting history” tutorial](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase-i)

It’s a tricky concept. To help you out, I went ahead & did a bit of rebasing of your commits myself. I’ll open another PR for this exact same work, and you can compare & use it as an example.

**Comment by jeffbyrnes** [comment]
And there we are. Please feel free to take a look! All credit to you, I just moved things around and cleaned things up a bit, expanded on the commit messages, and broke the version bump out to its own commit.

An aside to that, in most projects it’s considered best practice to let the maintainer handle changing the version number, updating the docs, and releasing said new version through whatever means they use. That said, here, it seems alright.

**Pull request closed without merging**